### PR TITLE
make Developer email optional as it is not obligatory and some people…

### PR DESCRIPTION
**Pull request by 0lejk4**
References #291 
As stated in issue this field should be optional. If i am misunderstanding something about its hidden meaning feel free to comment below. This fix should be great for my as i don`t want to expose my email when writing developer in build file

### CODE REVIEW

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=develop&zinc=develop&io=develop&librarymanagement=pull/292/head&util=develop&website=develop&deploy_to_bintray=false&run_sbt_community=false&verbose_dbuild=false&community_verbose=false">this link.</A>

**Comment by typesafe-tools** [comment]
A validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1455/console">is in progress...</a>

**Comment by 0lejk4** [comment]
Build failing in CI while locally everything was ok :( I guess i am missing something.

**Comment by typesafe-tools** [comment]
[//]: # (Refs: develop-develop-develop-pull/292/head-develop)
The validator <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1455/">has checked</A> the following projects, tested using dbuild, projects built on top of each other.

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/tree/develop">develop             </A> | sbt/sbt@354f85cecba4d5a165174e1471458940c523eda9
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/develop">develop             </A> | sbt/zinc@841966fbc699f5bd47c62f2bd6a752562adb82e4
  io                | <A HREF="https://github.com/sbt/io/tree/develop">develop             </A> | sbt/io@0ae81420bab2ed35a64c1691ac206efc323c9973
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/pull/292">pull/292/head       </A> | sbt/librarymanagement@0878b9db14ffaca87fdfcdee21d74c7b4b49a3a9
  util              | <A HREF="https://github.com/sbt/util/tree/develop">develop             </A> | sbt/util@48574c204117a91a47d10e19c52826ec4e5195c3

:x: The result is: ***FAILED***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild?sbt=develop&zinc=develop&io=develop&librarymanagement=pull/292/head&util=develop&deploy_to_bintray=false&run_sbt_community=false">(restart)</A>

**Comment by eed3si9n** [comment]
Unfortunately CI error is legit. It's failing on the binary compatibility check, which we cannot break during sbt 1.x life.

```
[error]  * method withEmail(java.lang.String)sbt.librarymanagement.Developer in class sbt.librarymanagement.Developer's type is different in current version, where it is (scala.Option)sbt.librarymanagement.Developer instead of (java.lang.String)sbt.librarymanagement.Developer
[error]    filter with: ProblemFilters.exclude[IncompatibleMethTypeProblem]("sbt.librarymanagement.Developer.withEmail")
[error]  * method email()java.lang.String in class sbt.librarymanagement.Developer has a different result type in current version, where it is scala.Option rather than java.lang.String
[error]    filter with: ProblemFilters.exclude[IncompatibleResultTypeProblem]("sbt.librarymanagement.Developer.email")
[error]  * method this(java.lang.String,java.lang.String,java.lang.String,java.net.URL)Unit in class sbt.librarymanagement.Developer's type is different in current version, where it is (java.lang.String,java.lang.String,scala.Option,java.net.URL)Unit instead of (java.lang.String,java.lang.String,java.lang.String,java.net.URL)Unit
[error]    filter with: ProblemFilters.exclude[IncompatibleMethTypeProblem]("sbt.librarymanagement.Developer.this")
[error]  * method apply(java.lang.String,java.lang.String,java.lang.String,java.net.URL)sbt.librarymanagement.Developer in object sbt.librarymanagement.Developer's type is different in current version, where it is (java.lang.String,java.lang.String,scala.Option,java.net.URL)sbt.librarymanagement.Developer instead of (java.lang.String,java.lang.String,java.lang.String,java.net.URL)sbt.librarymanagement.Developer
[error]    filter with: ProblemFilters.exclude[IncompatibleMethTypeProblem]("sbt.librarymanagement.Developer.apply")
...
[error] java.lang.RuntimeException: librarymanagement-core: Binary compatibility check failed!

```

File: ivy/src/main/scala/sbt/internal/librarymanagement/MakePom.scala [hunk 1]
```diff
@@ -206,8 +206,13 @@ class MakePom(val log: Logger) {
             <developer>
               <id>{ developer.id }</id>
               <name>{ developer.name }</name>
-              <email>{ developer.email }</email>
               <url>{ developer.url }</url>
+              {
+                developer.email match {
```
Comment (eed3si9n) [inline ivy/src/main/scala/sbt/internal/librarymanagement/MakePom.scala:211]: Maybe we can do something like if email is `""` make it `NodeSeq.Empty`?

**Review by eed3si9n — CHANGES_REQUESTED** [review]

File: ivy/src/main/scala/sbt/internal/librarymanagement/MakePom.scala (hunk 1 above)
Comment (0lejk4) [reply ivy/src/main/scala/sbt/internal/librarymanagement/MakePom.scala:211]: seems legit

Comment (0lejk4) [reply ivy/src/main/scala/sbt/internal/librarymanagement/MakePom.scala:211]: Should we also do same for null case?

**Comment by typesafe-tools** [comment]
A validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1463/console">is in progress...</a>

**Comment by typesafe-tools** [comment]
[//]: # (Refs: develop-develop-develop-pull/292/head-develop)
The validator <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/1463/">has checked</A> the following projects, tested using dbuild, projects built on top of each other.

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/tree/develop">develop             </A> | sbt/sbt@354f85cecba4d5a165174e1471458940c523eda9
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/develop">develop             </A> | sbt/zinc@35ebab05fd1b525fcd0bb5066a5424402b1a2e73
  io                | <A HREF="https://github.com/sbt/io/tree/develop">develop             </A> | sbt/io@0ae81420bab2ed35a64c1691ac206efc323c9973
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/pull/292">pull/292/head       </A> | sbt/librarymanagement@a4e5a2e299a43da1c1bcee9713f23c2b6fd23e91
  util              | <A HREF="https://github.com/sbt/util/tree/develop">develop             </A> | sbt/util@48574c204117a91a47d10e19c52826ec4e5195c3

:x: The result is: ***FAILED***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild?sbt=develop&zinc=develop&io=develop&librarymanagement=pull/292/head&util=develop&deploy_to_bintray=false&run_sbt_community=false">(restart)</A>

**Comment by 0lejk4** [comment]
Reworked without changing entity. Now it does not insert email tag if field is empty or null. If everything OK we can leave such behavior but in next 1.(x + 1) sbt release Developer schema should be changed to have optional email field.

**Review by eed3si9n — APPROVED** [review]
Thanks @0lejk4

**Pull request merged**